src/game_logic/ability_casting.rs

```diff
@@ -105,13 +112,144 @@ impl<'w, 's> AbilityCastingInterface<'w, 's> {
     }
 }
 
+/// Spawns a CastRequest entity for each UseAbility command
+fn request_ability_cast(mut commands: Commands, mut game_commands: MessageReader<GameCommand>) {
+    for command in game_commands.read() {
+        #[expect(irrefutable_let_patterns, reason = "We only have UseAbility currently")]
+        if let GameCommandKind::UseAbility(use_ability) = &command.kind {
+            commands.spawn(use_ability.clone());
+        }
+    }
+}
+
+/// Checks if the ability is on cooldown
+fn check_ability_cooldowns(
+    cast_requests: Query<(Entity, &UseAbility), Without<CastFailed<AbilityCooldown>>>,
+    has_cooldown: Query<Has<Cooldown>>,
+    mut commands: Commands,
+) {
+    for (req_e, use_ability) in cast_requests.iter() {
+        if has_cooldown.get(use_ability.ability_e).unwrap_or(false) {
+            commands
+                .entity(req_e)
+                .insert(CastFailed::<AbilityCooldown>::default());
+        }
+    }
+}
+
+#[derive(Debug, Clone, Reflect)]
+pub struct SlotCooldown;
+
+#[derive(Debug, Clone, Reflect)]
+pub struct SlotRequirement;
+
+/// Checks if the slot is on cooldown
+fn check_slot_cooldowns_real(
```

Comment:
```suggestion
fn check_slot_cooldowns(
```
+ proper rename in used places